### PR TITLE
nixd/Controller: reply `shutdown`

### PR DESCRIPTION
**Pull request by inclyc**

### CODE REVIEW

**Comment by inclyc** [comment]
`shutdown` was ignored before this commit. @Atemu could you confirm whether this PR fixes https://github.com/nix-community/nixd/issues/499  ?

**Review by Atemu — APPROVED** [review]
Nixd now shuts down "cleanly" from eglot's perspective, thank you!

The exit signal is 9; not sure whether that's expected or not.

**Comment by inclyc** [comment]
> The exit signal is 9; 

What did you mean by "exit signal"?

**Comment by Atemu** [comment]
Sorry, I meant exit code. it's just what eglot reports to me.

**Comment by inclyc** [comment]
> Sorry, I meant exit code. it's just what eglot reports to me. 

No, the exit code should be 0 though. How about paste some logs here then?

File: nixd/lib/Controller/LifeTime.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 /// [Server Lifecycle]:
 /// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#lifeCycleMessages
 
+#include "lspserver/Protocol.h"
```
Comment (inclyc) [inline nixd/lib/Controller/LifeTime.cpp:6]: canonicalize this header.

**Comment by Atemu** [comment]
Actually, I have misread: It says "status 9", not "exit code":

```
[eglot] Asking EGLOT (nixfiles/(nix-mode)) politely to terminate
[jsonrpc] Server exited with status 9
```

It's the same status code when I exit haskell-language-server, so it's probably correct.

Getting logs is surprisingly hard because eglot justifyably clears all log buffers upon successful shutdown.

**Pull request merged**